jobs/export_delete/worker.go

```diff
@@ -7,132 +7,60 @@ import (
-			mlog.String("worker", w.name),
-			mlog.String("job_id", job.Id),
-			mlog.String("error", err.Error()))
-		return
-	} else if !claimed {
-		return
-	}
-
-	exportPath := *w.app.Config().ExportSettings.Directory
-	retentionTime := time.Duration(*w.app.Config().ExportSettings.RetentionDays) * 24 * time.Hour
-	exports, appErr := w.app.ListDirectory(exportPath)
-	if appErr != nil {
-		w.setJobError(job, appErr)
-		return
+func MakeWorker(jobServer *jobs.JobServer, app AppIface) model.Worker {
+	isEnabled := func(cfg *model.Config) bool {
+		return *cfg.ExportSettings.Directory != "" && *cfg.ExportSettings.RetentionDays > 0
 	}
-
-	var hasErrs bool
-	for i := range exports {
-		filename := filepath.Base(exports[i])
-		modTime, appErr := w.app.FileModTime(filepath.Join(exportPath, filename))
+	execute := func(job *model.Job) error {
+		exportPath := *app.Config().ExportSettings.Directory
+		retentionTime := time.Duration(*app.Config().ExportSettings.RetentionDays) * 24 * time.Hour
+		exports, appErr := app.ListDirectory(exportPath)
 		if appErr != nil {
-			mlog.Debug("Worker: Failed to get file modification time",
-				mlog.Err(appErr), mlog.String("export", exports[i]))
-			hasErrs = true
-			continue
+			return appErr
 		}
 
-		if time.Now().After(modTime.Add(retentionTime)) {
-			// remove file data from storage.
-			if appErr := w.app.RemoveFile(exports[i]); appErr != nil {
-				mlog.Debug("Worker: Failed to remove file",
+		var hasErrs bool
+		for i := range exports {
+			filename := filepath.Base(exports[i])
+			modTime, appErr := app.FileModTime(filepath.Join(exportPath, filename))
+			if appErr != nil {
+				mlog.Debug("Worker: Failed to get file modification time",
 					mlog.Err(appErr), mlog.String("export", exports[i]))
 				hasErrs = true
 				continue
 			}
-		}
-	}
-
-	if hasErrs {
-		mlog.Warn("Worker: errors occurred")
-	}
 
-	mlog.Info("Worker: Job is complete", mlog.String("worker", w.name), mlog.String("job_id", job.Id))
-	w.setJobSuccess(job)
-}
-
-func (w *ExportDeleteWorker) setJobSuccess(job *model.Job) {
-	if err := w.app.Srv().Jobs.SetJobSuccess(job); err != nil {
-		mlog.Error("Worker: Failed to set success for job", mlog.String("worker", w.name), mlog.String("job_id", job.Id), mlog.String("error", err.Error()))
-		w.setJobError(job, err)
-	}
-}
+			if time.Now().After(modTime.Add(retentionTime)) {
+				// remove file data from storage.
+				if appErr := app.RemoveFile(exports[i]); appErr != nil {
+					mlog.Debug("Worker: Failed to remove file",
+						mlog.Err(appErr), mlog.String("export", exports[i]))
+					hasErrs = true
+					continue
+				}
+			}
+		}
 
-func (w *ExportDeleteWorker) setJobError(job *model.Job, appError *model.AppError) {
-	if err := w.app.Srv().Jobs.SetJobError(job, appError); err != nil {
-		mlog.Error("Worker: Failed to set job error", mlog.String("worker", w.name), mlog.String("job_id", job.Id), mlog.String("error", err.Error()))
+		if hasErrs {
+			mlog.Warn("Worker: errors occurred")
```

Comment:
Can we add some context to this warning? Worker name at least.  You could also collect the errors in a `merror.MError` and output them all here.

Reply:
This was logging the error just before, but I added your suggestion there of using multierror. See the commit: 59db72de58fe06b2c0ec0b4e6891f25c9c5df653